## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 475077`
Commit: `7d20394eab8b95e797c33359f300fa2f64414cbc`

---
### `x86_64-linux`
<details>
  <summary>:x: 15 packages failed to build:</summary>
  <ul>
    <li>hyprlandPlugins.borders-plus-plus</li>
    <li>hyprlandPlugins.csgo-vulkan-fix</li>
    <li>hyprlandPlugins.hy3</li>
    <li>hyprlandPlugins.hypr-darkwindow</li>
    <li>hyprlandPlugins.hypr-dynamic-cursors</li>
    <li>hyprlandPlugins.hyprbars</li>
    <li>hyprlandPlugins.hyprexpo</li>
    <li>hyprlandPlugins.hyprfocus</li>
    <li>hyprlandPlugins.hyprgrass</li>
    <li>hyprlandPlugins.hyprscrolling</li>
    <li>hyprlandPlugins.hyprspace</li>
    <li>hyprlandPlugins.hyprsplit</li>
    <li>hyprlandPlugins.hyprtrails</li>
    <li>hyprlandPlugins.hyprwinwrap</li>
    <li>hyprlandPlugins.xtra-dispatchers</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>grimblast</li>
    <li>hdrop</li>
    <li>hyprland</li>
    <li>hyprland.dev</li>
    <li>hyprland.man</li>
    <li>hyprpanel</li>
    <li>hyprshade</li>
    <li>hyprshade.dist</li>
    <li>hyprshell</li>
    <li>hyprshot</li>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>xdg-desktop-portal-hyprland</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>hyprlandPlugins.borders-plus-plus</summary>
<pre>Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j32 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[ 20%] Building CXX object CMakeFiles/borders-plus-plus.dir/BorderppPassElement.cpp.o
[ 40%] Building CXX object CMakeFiles/borders-plus-plus.dir/borderDeco.cpp.o
[ 60%] Building CXX object CMakeFiles/borders-plus-plus.dir/main.cpp.o
[ 80%] Building CXX object CMakeFiles/borders-plus-plus.dir/CMakeFiles/4.1.2/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
/build/borders-plus-plus/main.cpp:7:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
    7 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/build/borders-plus-plus/borderDeco.cpp:4:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
    4 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/borders-plus-plus.dir/build.make:93: CMakeFiles/borders-plus-plus.dir/borderDeco.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/borders-plus-plus.dir/build.make:121: CMakeFiles/borders-plus-plus.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/borders-plus-plus.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.csgo-vulkan-fix</summary>
<pre>    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR


-- Build files have been written to: /build/csgo-vulkan-fix/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j32 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[ 33%] Building CXX object CMakeFiles/csgo-vulkan-fix.dir/CMakeFiles/4.1.2/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
[ 66%] Building CXX object CMakeFiles/csgo-vulkan-fix.dir/main.cpp.o
/build/csgo-vulkan-fix/main.cpp:6:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
    6 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/csgo-vulkan-fix.dir/build.make:93: CMakeFiles/csgo-vulkan-fix.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/csgo-vulkan-fix.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hy3</summary>
<pre>                 from /build/source/src/main.cpp:9:
/build/source/src/Hy3Node.hpp:10:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
   10 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/hy3.dir/build.make:107: CMakeFiles/hy3.dir/src/Hy3Layout.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /build/source/src/Hy3Layout.hpp:32,
                 from /build/source/src/Hy3Node.cpp:12:
/build/source/src/Hy3Node.hpp:10:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
   10 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/hy3.dir/build.make:135: CMakeFiles/hy3.dir/src/TabGroup.cpp.o] Error 1
make[2]: *** [CMakeFiles/hy3.dir/build.make:93: CMakeFiles/hy3.dir/src/dispatchers.cpp.o] Error 1
make[2]: *** [CMakeFiles/hy3.dir/build.make:149: CMakeFiles/hy3.dir/src/SelectionHook.cpp.o] Error 1
compilation terminated.
make[2]: *** [CMakeFiles/hy3.dir/build.make:121: CMakeFiles/hy3.dir/src/Hy3Node.cpp.o] Error 1
make[2]: *** [CMakeFiles/hy3.dir/build.make:79: CMakeFiles/hy3.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hy3.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hypr-darkwindow</summary>
<pre>   25 |             Debug::log(ERR, &quot;Failed to edit GLSL Code, no gl_FragColor:\n{}&quot;, source);
      |             ^~~~~
src/Helpers.cpp:25:24: error: &#x27;ERR&#x27; was not declared in this scope; did you mean &#x27;Log::ERR&#x27;?
   25 |             Debug::log(ERR, &quot;Failed to edit GLSL Code, no gl_FragColor:\n{}&quot;, source);
      |                        ^~~
      |                        Log::ERR
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/debug/log/Logger.hpp:57:54: note: &#x27;Log::ERR&#x27; declared here
   57 |     inline constexpr const Hyprutils::CLI::eLogLevel ERR   = Hyprutils::CLI::LOG_ERR;
      |                                                      ^~~
src/Helpers.cpp:33:9: error: &#x27;Debug&#x27; has not been declared
   33 |         Debug::log(ERR, &quot;Failed to edit GLSL Code, no main function: {}&quot;, source);
      |         ^~~~~
src/Helpers.cpp:33:20: error: &#x27;ERR&#x27; was not declared in this scope; did you mean &#x27;Log::ERR&#x27;?
   33 |         Debug::log(ERR, &quot;Failed to edit GLSL Code, no main function: {}&quot;, source);
      |                    ^~~
      |                    Log::ERR
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/debug/log/Logger.hpp:57:54: note: &#x27;Log::ERR&#x27; declared here
   57 |     inline constexpr const Hyprutils::CLI::eLogLevel ERR   = Hyprutils::CLI::LOG_ERR;
      |                                                      ^~~
make: *** [Makefile:23: out/obj/Helpers.o] Error 1</pre>
</details>
<details>
<summary>hyprlandPlugins.hypr-dynamic-cursors</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/ywgba7vqr1qq429c7jxa3zgkl4yn8964-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
g++ -Wall --no-gnu-unique -fPIC -std=c++26 -g -I/nix/store/22rmj7c1knxadi0ks56q2x4s3ym3b7kn-libxcb-wm-0.4.2-dev/include/src -I/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include -I/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/protocols -I/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland -I/nix/store/sswjyx3vgl2dv2xl2sqw2x2n48lqj5r8-aquamarine-0.10.0-dev/include -I/nix/store/kx95a2drx2mgkxb61k65xlvp0dcbpsly-cairo-1.18.4-dev/include/cairo -I/nix/store/zykjcxzj7mg1x325xxfnrymf50daqvgb-freetype-2.13.3-dev/include/freetype2 -I/nix/store/zykjcxzj7mg1x325xxfnrymf50daqvgb-freetype-2.13.3-dev/include -I/nix/store/pw03xfrnb6a7nz6ivq6jijsx0n52afnp-pixman-0.46.4/include/pixman-1 -I/nix/store/p7c0w266jlbzwh84qw6d5g5glbr8mzyp-libxcb-1.17.0-dev/include -I/nix/store/nc627sydkid97k91c6rxih3abz10pag0-hyprcursor-0.1.13-dev/include -I/nix/store/gav3x93afbhn9gjb937dikcvvm52jh4v-hyprgraphics-0.4.0-dev/include -I/nix/store/ympz5ib3vl3kbj4gx99byf765f87w9il-hyprlang-0.6.7-dev/include -I/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include -I/nix/store/b608h3ylrhrwpi9bc0y0014jf7j03pzl-libglvnd-1.7.0-dev/include -I/nix/store/vb7mcmb8sq7ab2x7d0ihph62falar66q-libdrm-2.4.128-dev/include -I/nix/store/vb7mcmb8sq7ab2x7d0ihph62falar66q-libdrm-2.4.128-dev/include/libdrm -I/nix/store/06fhm6afbwxwq1cdr0fgmigk7zm3ipdg-libinput-1.29.2-dev/include -I/nix/store/crgawvhg4k8lh37mp40jiqg11rdyv5g0-libxkbcommon-1.11.0-dev/include -I/nix/store/qxvgjcqizyz0rhrwlm49i0ys2gcgavf0-wayland-1.24.0-dev/include -I/nix/store/1wakh35bcwla481xfd999b1mdm40rmb1-libxcb-errors-1.0.1-dev/include -I/nix/store/22rmj7c1knxadi0ks56q2x4s3ym3b7kn-libxcb-wm-0.4.2-dev/include -c src/cursor.cpp -o out/cursor.o
In file included from src/cursor.cpp:12:
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/managers/CursorManager.hpp:10:10: fatal error: managers/eventLoop/EventLoopManager.hpp: No such file or directory
   10 | #include &quot;managers/eventLoop/EventLoopManager.hpp&quot;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:21: out/cursor.o] Error 1</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprbars</summary>
<pre>In file included from /build/hyprbars/barDeco.cpp:1:
/build/hyprbars/barDeco.hpp:9:10: fatal error: hyprland/src/desktop/WindowRule.hpp: No such file or directory
    9 | #include &lt;hyprland/src/desktop/WindowRule.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /build/hyprbars/BarPassElement.cpp:3:
/build/hyprbars/barDeco.hpp:9:10: fatal error: hyprland/src/desktop/WindowRule.hpp: No such file or directory
    9 | #include &lt;hyprland/src/desktop/WindowRule.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/build/hyprbars/main.cpp:7:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
    7 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/hyprbars.dir/build.make:121: CMakeFiles/hyprbars.dir/main.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/hyprbars.dir/build.make:93: CMakeFiles/hyprbars.dir/barDeco.cpp.o] Error 1
make[2]: *** [CMakeFiles/hyprbars.dir/build.make:79: CMakeFiles/hyprbars.dir/BarPassElement.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprbars.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprexpo</summary>
<pre>/build/hyprexpo/overview.cpp:50:9: error: &#x27;Debug&#x27; has not been declared
   50 |         Debug::log(ERR, &quot;[he] invalid workspace_method&quot;);
      |         ^~~~~
/build/hyprexpo/overview.cpp:50:20: error: &#x27;ERR&#x27; was not declared in this scope; did you mean &#x27;Log::ERR&#x27;?
   50 |         Debug::log(ERR, &quot;[he] invalid workspace_method&quot;);
      |                    ^~~
      |                    Log::ERR
In file included from /nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/macros.hpp:9,
                 from /nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/helpers/Color.hpp:5,
                 from /nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/plugins/PluginAPI.hpp:23,
                 from /build/hyprexpo/globals.hpp:3,
                 from /build/hyprexpo/overview.hpp:5,
                 from /build/hyprexpo/overview.cpp:1:
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/debug/log/Logger.hpp:57:54: note: &#x27;Log::ERR&#x27; declared here
   57 |     inline constexpr const Hyprutils::CLI::eLogLevel ERR   = Hyprutils::CLI::LOG_ERR;
      |                                                      ^~~
make[2]: *** [CMakeFiles/hyprexpo.dir/build.make:79: CMakeFiles/hyprexpo.dir/ExpoGesture.cpp.o] Error 1
make[2]: *** [CMakeFiles/hyprexpo.dir/build.make:135: CMakeFiles/hyprexpo.dir/overview.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprexpo.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprfocus</summary>
<pre>
-- Build files have been written to: /build/hyprfocus/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j32 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[ 33%] Building CXX object CMakeFiles/hyprfocus.dir/main.cpp.o
[ 66%] Building CXX object CMakeFiles/hyprfocus.dir/CMakeFiles/4.1.2/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
/build/hyprfocus/main.cpp: In lambda function:
/build/hyprfocus/main.cpp:58:28: error: &#x27;class CCompositor&#x27; has no member named &#x27;updateWindowAnimatedDecorationValues&#x27;; did you mean &#x27;updateAllWindowsAnimatedDecorationValues&#x27;?
   58 |             g_pCompositor-&gt;updateWindowAnimatedDecorationValues(w.lock());
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                            updateAllWindowsAnimatedDecorationValues
/build/hyprfocus/main.cpp:59:16: error: &#x27;class Desktop::View::CWindow&#x27; has no member named &#x27;updateDynamicRules&#x27;
   59 |             w-&gt;updateDynamicRules();
      |                ^~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/hyprfocus.dir/build.make:93: CMakeFiles/hyprfocus.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprfocus.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprgrass</summary>
<pre>../src/TouchVisualizer.cpp:70:59: error: &#x27;class CCompositor&#x27; has no member named &#x27;m_lastMonitor&#x27;
   70 |     g_pCompositor-&gt;scheduleFrameForMonitor(g_pCompositor-&gt;m_lastMonitor.lock());
      |                                                           ^~~~~~~~~~~~~
../src/TouchVisualizer.cpp: In member function &#x27;void Visualizer::onTouchMotion(ITouch::SMotionEvent)&#x27;:
../src/TouchVisualizer.cpp:74:57: error: &#x27;class CCompositor&#x27; has no member named &#x27;m_lastMonitor&#x27;
   74 |     auto mon                           = g_pCompositor-&gt;m_lastMonitor.lock();
      |                                                         ^~~~~~~~~~~~~
../src/TouchVisualizer.cpp:75:100: error: no match for &#x27;operator=&#x27; (operand types are &#x27;std::unordered_map&lt;int, FingerPos&gt;::mapped_type&#x27; {aka &#x27;FingerPos&#x27;} and &#x27;&lt;brace-enclosed initializer list&gt;&#x27;)
   75 |     this-&gt;finger_positions[ev.touchID] = {ev.pos * mon-&gt;m_pixelSize + mon-&gt;m_position, std::nullopt};
      |                                                                                                    ^
../src/TouchVisualizer.cpp:75:100: note: there are 2 candidates
In file included from ../src/TouchVisualizer.cpp:1:
../src/TouchVisualizer.hpp:5:8: note: candidate 1: &#x27;constexpr FingerPos&amp; FingerPos::operator=(const FingerPos&amp;)&#x27;
    5 | struct FingerPos {
      |        ^~~~~~~~~
../src/TouchVisualizer.hpp:5:8: note: no known conversion for argument 1 from &#x27;&lt;brace-enclosed initializer list&gt;&#x27; to &#x27;const FingerPos&amp;&#x27;
../src/TouchVisualizer.hpp:5:8: note: candidate 2: &#x27;constexpr FingerPos&amp; FingerPos::operator=(FingerPos&amp;&amp;)&#x27;
../src/TouchVisualizer.hpp:5:8: note: no known conversion for argument 1 from &#x27;&lt;brace-enclosed initializer list&gt;&#x27; to &#x27;FingerPos&amp;&amp;&#x27;
[11/14] Compiling C++ object src/gestures/test/test-gestures.p/test.cpp.o
ninja: build stopped: subcommand failed.</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprscrolling</summary>
<pre>   78 |         CBox(const double d) {
      |         ^~~~
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:78:9: note: candidate expects 1 argument, 2 provided
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:70:9: note: candidate 3: &#x27;Hyprutils::Math::CBox::CBox()&#x27;
   70 |         CBox() {
      |         ^~~~
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:70:9: note: candidate expects 0 arguments, 2 provided
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:61:9: note: candidate 4: &#x27;Hyprutils::Math::CBox::CBox(double, double, double, double)&#x27;
   61 |         CBox(double x_, double y_, double w_, double h_) {
      |         ^~~~
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:61:9: note: candidate expects 4 arguments, 2 provided
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:52:11: note: candidate 5: &#x27;constexpr Hyprutils::Math::CBox::CBox(const Hyprutils::Math::CBox&amp;)&#x27;
   52 |     class CBox {
      |           ^~~~
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:52:11: note: candidate expects 1 argument, 2 provided
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:52:11: note: candidate 6: &#x27;constexpr Hyprutils::Math::CBox::CBox(Hyprutils::Math::CBox&amp;&amp;)&#x27;
/nix/store/2prcdg59nziibq9vvpxgb2wli6mvykw9-hyprutils-0.11.0-dev/include/hyprutils/math/Box.hpp:52:11: note: candidate expects 1 argument, 2 provided
make[2]: *** [CMakeFiles/hyprscrolling.dir/build.make:79: CMakeFiles/hyprscrolling.dir/Scrolling.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprscrolling.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprspace</summary>
<pre>  515 |     HyprlandAPI::addDispatcher(pHandle, &quot;overview:toggle&quot;, ::dispatchToggleOverview);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/plugins/PluginSystem.hpp:6,
                 from src/main.cpp:1:
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/plugins/PluginAPI.hpp:248:33: note: declared here
  248 |     APICALL [[deprecated]] bool addDispatcher(HANDLE handle, const std::string&amp; name, std::function&lt;void(std::string)&gt; handler);
      |                                 ^~~~~~~~~~~~~
src/main.cpp:516:31: warning: &#x27;bool HyprlandAPI::addDispatcher(void*, const std::string&amp;, std::function&lt;void(std::__cxx11::basic_string&lt;char&gt;)&gt;)&#x27; is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
  516 |     HyprlandAPI::addDispatcher(pHandle, &quot;overview:open&quot;, ::dispatchOpenOverview);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/plugins/PluginAPI.hpp:248:33: note: declared here
  248 |     APICALL [[deprecated]] bool addDispatcher(HANDLE handle, const std::string&amp; name, std::function&lt;void(std::string)&gt; handler);
      |                                 ^~~~~~~~~~~~~
src/main.cpp:517:31: warning: &#x27;bool HyprlandAPI::addDispatcher(void*, const std::string&amp;, std::function&lt;void(std::__cxx11::basic_string&lt;char&gt;)&gt;)&#x27; is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc-15.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
  517 |     HyprlandAPI::addDispatcher(pHandle, &quot;overview:close&quot;, ::dispatchCloseOverview);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/27d7xv4xnpy4y881jh40yab5awzrds10-hyprland-0.53.0-dev/include/hyprland/src/plugins/PluginAPI.hpp:248:33: note: declared here
  248 |     APICALL [[deprecated]] bool addDispatcher(HANDLE handle, const std::string&amp; name, std::function&lt;void(std::string)&gt; handler);
      |                                 ^~~~~~~~~~~~~
make: *** [Makefile:7: all] Error 1</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprsplit</summary>
<pre>  426 |     Debug::log(LOG, &quot;[hyprsplit] monitor removed {}&quot;, pMonitor-&gt;m_name);
      |                ^~~
      |                Log
../src/main.cpp: In function &#x27;PLUGIN_DESCRIPTION_INFO pluginInit(void*)&#x27;:
../src/main.cpp:485:5: error: &#x27;Debug&#x27; has not been declared
  485 |     Debug::log(LOG, &quot;[hyprsplit] plugin init&quot;);
      |     ^~~~~
../src/main.cpp:485:16: error: &#x27;LOG&#x27; was not declared in this scope; did you mean &#x27;Log&#x27;?
  485 |     Debug::log(LOG, &quot;[hyprsplit] plugin init&quot;);
      |                ^~~
      |                Log
../src/main.cpp: In function &#x27;void pluginExit()&#x27;:
../src/main.cpp:490:5: error: &#x27;Debug&#x27; has not been declared
  490 |     Debug::log(LOG, &quot;[hyprsplit] plugin exit&quot;);
      |     ^~~~~
../src/main.cpp:490:16: error: &#x27;LOG&#x27; was not declared in this scope; did you mean &#x27;Log&#x27;?
  490 |     Debug::log(LOG, &quot;[hyprsplit] plugin exit&quot;);
      |                ^~~
      |                Log
ninja: build stopped: subcommand failed.</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprtrails</summary>
<pre>Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j32 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[ 20%] Building CXX object CMakeFiles/hyprtrails.dir/TrailPassElement.cpp.o
[ 40%] Building CXX object CMakeFiles/hyprtrails.dir/CMakeFiles/4.1.2/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
[ 60%] Building CXX object CMakeFiles/hyprtrails.dir/main.cpp.o
[ 80%] Building CXX object CMakeFiles/hyprtrails.dir/trail.cpp.o
/build/hyprtrails/main.cpp:7:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
    7 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/build/hyprtrails/trail.cpp:4:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
    4 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/hyprtrails.dir/build.make:107: CMakeFiles/hyprtrails.dir/main.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/hyprtrails.dir/build.make:121: CMakeFiles/hyprtrails.dir/trail.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprtrails.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.hyprwinwrap</summary>
<pre>    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR


-- Build files have been written to: /build/hyprwinwrap/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j32 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[ 33%] Building CXX object CMakeFiles/hyprwinwrap.dir/CMakeFiles/4.1.2/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
[ 66%] Building CXX object CMakeFiles/hyprwinwrap.dir/main.cpp.o
/build/hyprwinwrap/main.cpp:12:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
   12 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/hyprwinwrap.dir/build.make:93: CMakeFiles/hyprwinwrap.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/hyprwinwrap.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
<details>
<summary>hyprlandPlugins.xtra-dispatchers</summary>
<pre>    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR


-- Build files have been written to: /build/xtra-dispatchers/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j32 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[ 33%] Building CXX object CMakeFiles/xtra-dispatchers.dir/CMakeFiles/4.1.2/CompilerIdCXX/CMakeCXXCompilerId.cpp.o
[ 66%] Building CXX object CMakeFiles/xtra-dispatchers.dir/main.cpp.o
/build/xtra-dispatchers/main.cpp:10:10: fatal error: hyprland/src/desktop/Window.hpp: No such file or directory
   10 | #include &lt;hyprland/src/desktop/Window.hpp&gt;
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/xtra-dispatchers.dir/build.make:93: CMakeFiles/xtra-dispatchers.dir/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/xtra-dispatchers.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/nixos/modules/programs/wayland/hyprland.nix
+++ b/nixos/modules/programs/wayland/hyprland.nix
@@ -14,7 +14,7 @@ in
   options.programs.hyprland = {
     enable = lib.mkEnableOption ''
       Hyprland, the dynamic tiling Wayland compositor that doesn't sacrifice on its looks.
-      You can manually launch Hyprland by executing {command}`Hyprland` on a TTY.
+      You can manually launch Hyprland by executing {command}`start-hyprland` on a TTY.
       A configuration file will be generated in {file}`~/.config/hypr/hyprland.conf`.
       See <https://wiki.hyprland.org> for more information'';
 
@@ -103,7 +103,7 @@ in
           hyprland = {
             prettyName = "Hyprland";
             comment = "Hyprland compositor managed by UWSM";
-            binPath = "/run/current-system/sw/bin/Hyprland";
+            binPath = "/run/current-system/sw/bin/start-hyprland";
           };
         };
       })

--- a/pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix
+++ b/pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix
@@ -13,13 +13,13 @@ let
             mkHyprlandPlugin,
           }:
           let
-            version = "0.52.0";
+            version = "0.53.0";
 
             hyprland-plugins-src = fetchFromGitHub {
               owner = "hyprwm";
               repo = "hyprland-plugins";
               tag = "v${version}";
-              hash = "sha256-hr53AWO96ooLCwS1a2v416eT1/aWQZmuQV0ULqhaBTY=";
+              hash = "sha256-BSRT1Uu1ot4WfMfZc6KW0nwpmt2xl9wpUqmH/JoMTfk=";
             };
           in
           mkHyprlandPlugin {

--- a/pkgs/by-name/hy/hyprland/info.json
+++ b/pkgs/by-name/hy/hyprland/info.json
@@ -1,7 +1,7 @@
 {
-  "branch": "v0.52.2-b",
-  "commit_hash": "386376400119dd46a767c9f8c8791fd22c7b6e61",
-  "commit_message": "[gha] Nix: update inputs",
-  "date": "2025-12-03",
-  "tag": "v0.52.2"
+  "branch": "main",
+  "commit_hash": "ea444c35bb23b6e34505ab6753e069de7801cc25",
+  "commit_message": "version: bump to 0.53.0",
+  "date": "2025-12-29",
+  "tag": "v0.53.0"
 }

--- a/pkgs/by-name/hy/hyprland/package.nix
+++ b/pkgs/by-name/hy/hyprland/package.nix
@@ -19,6 +19,7 @@
   hyprland-qtutils,
   hyprlang,
   hyprutils,
+  hyprwire,
   hyprwayland-scanner,
   libGL,
   libdrm,
@@ -91,14 +92,14 @@ assert assertMsg (
 
 customStdenv.mkDerivation (finalAttrs: {
   pname = "hyprland" + optionalString debug "-debug";
-  version = "0.52.2";
+  version = "0.53.0";
 
   src = fetchFromGitHub {
     owner = "hyprwm";
     repo = "hyprland";
     fetchSubmodules = true;
     tag = "v${finalAttrs.version}";
-    hash = "sha256-R2Hm7XbW8CTLEIeYCAlSQ3U5bFhn76FC17hEy/ws8EM=";
+    hash = "sha256-1jZK7hqNhQRqhj+2eb/JvnBoARxUgoVXKLSwp2RPmNQ=";
   };
 
   postPatch = ''
@@ -107,6 +108,7 @@ customStdenv.mkDerivation (finalAttrs: {
 
     # Remove extra @PREFIX@ to fix pkg-config paths
     sed -i "s#@PREFIX@/##g" hyprland.pc.in
+    sed -i "s#@PREFIX@/##g" example/hyprland.desktop.in
   '';
 
   # variables used by CMake, and shown in `hyprctl version`
@@ -127,6 +129,7 @@ customStdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [
     hyprwayland-scanner
+    hyprwire
     makeWrapper
     cmake
     # meson + ninja are used to build the hyprland-protocols submodule
@@ -187,13 +190,13 @@ customStdenv.mkDerivation (finalAttrs: {
   strictDeps = true;
 
   cmakeFlags = mapAttrsToList cmakeBool {
+    "BUILT_WITH_NIX" = true;
     "NO_XWAYLAND" = !enableXWayland;
     "NO_SYSTEMD" = !withSystemd;
     "CMAKE_DISABLE_PRECOMPILE_HEADERS" = true;
     "NO_UWSM" = true;
     "NO_HYPRPM" = true;
     "TRACY_ENABLE" = false;
-    "BUILD_HYPRTESTER" = true;
   };
 
   postInstall = ''

